### ISSUE TITLE
Commit should be pushed before creating tag

### ISSUE THREAD

**Issue by sdankel** [undergroundwires/bump-everywhere#18]
First of all, thank you for creating this! I love the idea of bundling all of this boilerplate activity into an easy-to-use action. 

My expectation is that the version bump in package.json and generated release notes should be included in the released version. However, it doesn't appear to be working that way. For example, in the last release of this project: 

https://github.com/undergroundwires/bump-everywhere/compare/1.3.1...1.4.0

You can see that it includes the release notes and version bump from the _previous_ release, rather than from 1.4.0. Looking at the code, I can see this is because the tag is created and pushed before the commit is made. I see two options:

1. Move the logic that creates and pushes the tag to after the commit is pushed or 
2. Retag the tag after the commit is pushed to point to the hash of the commit

I prefer option 1 because it means that if the action ever fails to push the commit, there is no need to clean up the tags. Happy to submit a PR for this if this proposal is accepted.

**Comment by zomars** [comment]
Agreed with @sdankel this is the only thing preventing us to adopt this into @calcom
